src/App.tsx

```diff
@@ -1,26 +1,217 @@
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setDeletingId(todoId);
+
+    return todoService
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        inputRef.current?.focus();
+      })
+      .catch(error => {
+        setTodos(todos);
+        setErrorMessage('Unable to delete a todo');
+        throw error;
+      })
+      .finally(() => {
+        setDeletingId(null);
+      });
+  };
+
+  const deleteTodos = (todoIds: number[]) => {
+    const promices = todoIds.map(todoId => todoService.deleteTodo(todoId));
+
+    return Promise.allSettled(promices).then(results => {
+      const errors = results.filter(result => result.status === 'rejected');
+      const successIds = results
+        .filter(result => result.status === 'fulfilled')
+        .map(result => result.value.id);
+
+      return { errors, successIds };
+    });
+  };
+
+  const deleteCompletedTodos = () => {
+    setLoading(true);
+
+    const completedTodos = todos.filter(todo => todo.completed);
+    const completedTodoIds = completedTodos.map(todo => todo.id);
+
+    deleteTodos(completedTodoIds)
+      .then(({ errors, successIds }) => {
+        if (errors.length > 0) {
+          setErrorMessage('Unable to delete a todo');
+        }
+
+        setTodos(currentTodos =>
+          currentTodos
+            .filter(todo => !successIds.includes(todo.id))
+            .filter(todo => !todo.completed),
+        );
+
+        setLoading(false);
+      })
+      .catch(() => {
+        setTodos(todos);
+        setLoading(false);
+      });
+  };
+
+  const handleFilter = (currentFilter: string) => {
+    setFilter(currentFilter);
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          addTodo={addTodo}
+          inputTitle={inputTitle}
+          setInputTitle={setInputTitle}
+          inputRef={inputRef}
+        />
+
+        <TodoList
+          filteredTodos={filteredTodos}
+          deleteTodo={deleteTodo}
+          deletingId={deletingId}
+          tempTodo={tempTodo}
+        />
+
+        {todos.length !== 0 && (
```

Comment:
Not fixed 
```suggestion
        {!!todos.length && (
```
<img width="546" alt="image" src="https://github.com/user-attachments/assets/daca160d-bcc5-4dd9-926e-469b38a2a083">